tests/petsc/petsc_snes_02.output.clang-16

```diff
@@ -0,0 +1,11 @@
+
+DEAL::
+--------------------------------------------------------
+An error occurred in file <petsc_snes.templates.h> in function
+    unsigned int dealii::PETScWrappers::NonlinearSolver<VectorType, MatrixType>::solve(VectorType &) [VectorType = VectorType, PMatrixType = MatrixType, AMatrixType = MatrixType]
+The violated condition was: 
+    residual
+Additional information: 
+    Please provide an implementation for the function "residual"
+--------------------------------------------------------
```

Comment:
This partially overlaps with the other PR. Since printing these is compiler-dependent, could you instead modify the test to print a specific message for this exception?

Reply:
The test can catch the notImplemented exception and print a special message. There is more of this compiler-specific output in the other tests I have added at https://github.com/dealii/dealii/pull/15276. Some of those  require to define new uniform exceptions across the various nonlinear and time-dependent solvers. @bangerth I will open an issue for that